docs/advanced/featureflags.md

```diff
@@ -0,0 +1,280 @@
+    
+    // mark as requiring the 'EXPERIMENTAL' flag
+    FeatureFlagSet.of(EXPERIMENTAL)
+));
+
+// MobEffect
+DeferredRegister<MobEffect> MOB_EFFECTS = DeferredRegister.create(Registries.MOB_EFFECT, "examplemod");
+DeferredHolder<MobEffect, ExperimentalMobEffect> EXPERIMENTAL_MOB_EEFECT = MOB_EFFECTS.register("experimental", registryName -> new ExperimentalMobEffect(MobEffectCategory.NEUTRAL, CommonColors.WHITE)
+    .requiredFeatures(EXPERIMENTAL) // mark as requiring the 'EXPERIMENTAL' flag
+);
+
+// Potion
+DeferredRegister<Potion> POTIONS = DeferredRegister.create(Registries.POTION, "examplemod");
+DeferredHolder<Potion, ExperimentalPotion> EXPERIMENTAL_POTION = POTIONS.register("experimental", registryName -> new ExperimentalPotion(registryName.toString(), new MobEffectInstance(EXPERIMENTAL_MOB_EEFECT))
+    .requiredFeatures(EXPERIMENTAL) // mark as requiring the 'EXPERIMENTAL' flag
+);
+```
+
+**Validating enabled status**
+
+In order to validate if features should be enabled or not, you must first acquire the set of enabled features. This can be retrieved in a variety of ways but the common and recommended method is `LevelReader#enabledFeatures`.  
+
+```java
+level.enabledFeatures(); // from a 'LevelReader' instance
+entity.level().enabledFeatures(); // from a 'Entity' instance
+
+// Client Side
+minecraft.getConnection().enabledFeatures();
+
+// Server Side
+server.getWorldData().enabledFeatures();
+```
+
+To validate if any arbitrary `FeatureFlagSet` is enabled, simply pass the enabled features to `FeatureFlagSet#isSubsetOf`.
+
+To validate if any given `FeatureElement` is enabled, simply invoke `FeatureElement#isEnabled`.
+
+:::note
+`ItemStack` has a speciality `isItemEnabled(FeatureFlagSet)` method, This is so that empty stacks are treated as enabled even if the required features for backing `Item` does not match the enabled features.
+
+It is recommended to make use of this speciality over `Item#isEnabled` where possible.
+:::
+
+```java
+requiredFeatures.isSubsetOf(enabledFeatures);
+featureElement.isEnabled(enabledFeatures);
+itemStack.isItemEnabled(enabledFeatures);
+```
+
+### Feature Packs
+
+Feature Packs are a type of pack which not only loads resources and/or data but also has the ability to toggle on a given set of Feature Flags.
+
+These flags are defined in the `pack.mcmeta` JSON file at the root of this pack, which follows the below format.
+```json5
+{
+    "features": [
+        "enabled": [
+            // Identifier of a Feature Flag to be enabled
+            // Must be a valid registered flag
+            "examplemod:experimental"
+        ]
+    ]
+}
+```
+
+There are a couple methods in which you can provide a feature pack to your users.
+
+Installing these packs differs depending on if its being installed into a single player world or a server.
+
+**External**
+
+External packs are provided to your users in datapack form.
+
+**Installing into a single player world**
```

Comment:
```suggestion
#### Installation in Singleplayer
```
Make this a H4 since it is a subsection of External, which would be a H3. Also add a new line after the heading.